confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java

```diff
@@ -76,33 +81,63 @@ private void setConsensusLayer() throws IOException {
                             conf.getConfigNodeConsensusProtocolClass())));
     consensusImpl.start();
 
-    // Build ratis group from user properties
-    LOGGER.info(
-        "Set ConfigNode consensus group {}...",
-        Arrays.toString(conf.getConfigNodeGroupAddressList()));
+    // Build consensus group from iotdb-confignode.properties
+    LOGGER.info("Set ConfigNode consensus group {}...", conf.getConfigNodeList());
     List<Peer> peerList = new ArrayList<>();
-    for (TEndPoint endpoint : conf.getConfigNodeGroupAddressList()) {
-      peerList.add(new Peer(consensusGroupId, endpoint));
+    for (TConfigNodeLocation configNodeLocation : conf.getConfigNodeList()) {
+      peerList.add(new Peer(consensusGroupId, configNodeLocation.getConsensusEndPoint()));
     }
     consensusImpl.addConsensusGroup(consensusGroupId, peerList);
+
+    // Apply ConfigNode if necessary
+    if (conf.isNeedApply()) {
+      TSStatus status =
+          SyncConfigNodeClientPool.getInstance()
+              .applyConfigNode(
+                  conf.getTargetConfigNode(),
+                  new TConfigNodeLocation(
+                      new TEndPoint(conf.getRpcAddress(), conf.getRpcPort()),
+                      new TEndPoint(conf.getRpcAddress(), conf.getConsensusPort())));
+      if (status.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        LOGGER.error(status.getMessage());
+        throw new IOException("Apply ConfigNode failed:");
+      }
+    }
+  }
+
+  /**
+   * Apply new ConfigNode Peer into PartitionRegion
+   *
+   * @param applyConfigNodeReq ApplyConfigNodeReq
+   * @return True if successfully addPeer. False if another ConfigNode is being added to the
+   *     PartitionRegion
+   */
+  public boolean applyConfigNodePeer(ApplyConfigNodeReq applyConfigNodeReq) {
```

Comment:
Add `add` or something to this function name? Otherwise the apply would be ambiguous

Reply:
Fixed.